### ISSUE TITLE
Discussion: Video tutorial series

### ISSUE THREAD

**Issue by amilajack** [electron-react-boilerplate/electron-react-boilerplate#518]
I'm thinking of making a video tutorial series for this repo. I'm thinking of covering the following topics:

* Getting Started
  * IDE/text editor setup
  * Project Directory Structure walkthrough
  * Configuration explanation
    * Webpack, babel, eslint, flow, electron, npm scripts, redux
  * Creating Components
  * Adding env variables to dev and prod
  * Routing
  * Managing state with redux
  * Testing with spectron, enzyme, and jest
  * Creating a todo list sample app from scratch using the boilerplate
* Advanced 
  * IPC
  * Debugging main/renderer processes

I'm sure there are video tutorial series out there for these topics but based on the issues/questions that have asked and made, I think explaining these topics in the context of this project would help significantly, especially for those who are new to the JS ecosystem.

**Comment by AlexFrazer** [comment]
wouldn't IDE setup be a bit opinionated?

**Comment by amilajack** [comment]
We already have a [list of plugins that we recommend](https://github.com/chentsulin/electron-react-boilerplate#editor-configuration). I'm thinking of covering a wide variety of tools and IDE's including:

* vim
* sublime
* atom
* vs code

**Comment by stefanKuijers** [comment]
Great idea!

1) It would be good if you suggest some good starter react resources so that you don't have to teach react it self. It might be good to focus on an overview of the technologies and only go into detail about the boilerplate itself. 

2) Creating a good screencast takes time. Only put it in if you are confident that there are not many big changes coming up as the screencast will become useless quite quickly. That would be a shame.

**Comment by amilajack** [comment]
> It would be good if you suggest some good starter react resources

Never even thought of this. Great idea!

**Comment by m-brantheim** [comment]
Sounds like an excellent idea! I am a bit confused myself on how to set up the state management with Redux, it is a little bit hard to get started with the boilerplate for a beginner like me, with so much stuff going on so I think a tutorial could help.

**Comment by amilajack** [comment]
js fatigue is one of my growning concerns with this boilerplate. Its actually the reason why I initially thought of creating a video tutorial series.

**Comment by jefffriesen** [comment]
@m-brantheim If you have't seen it, Dan Abranov did an excellent, free video series on Redux:
https://egghead.io/courses/getting-started-with-redux

Since there are quite a few React and Redux resources out there I would vote for emphasizing Electron-specific tutorials. Then link to good react and redux resources like the one above.

**Comment by amilajack** [comment]
@m-brantheim Yup! I've seen it. I was planning on electron specific tutorials. Also there seem to be a lot of questions regarding how to create new components and add them so I'll probably cover that.

**Comment by rumanbsl** [comment]
Even a written tutorial would suffice, I am scratching my head on in package.json scripts.
for example
`"hot-updates-server": "cross-env NODE_ENV=development node --trace-warnings -r babel-register ./node_modules/webpack-dev-server/bin/webpack-dev-server --config webpack.config.renderer.dev.js",`

specifically `node --trace-warnings -r babel-register` this part. How does it go along with subsequent webpack call

**Comment by amilajack** [comment]
@rumanbsl I just added `npm scripts` to the topics that I intend to cover

**Comment by rumanbsl** [comment]
@amilajack would it be possible to share.

**Comment by amilajack** [comment]
Are you referring to the video tutorials? I haven't worked on them yet so I can't show them.

**Comment by JPaveg** [comment]
I agree with all the topics you've listed with the exception of react/ redux (unless they're brief). No need to do unnecessary work :)

Very excited for the videos as the main reason I haven't actually used this project in production is because I'm quite new to es6+ js and the related ecosystem. I'd like to fully understand exactly what the boilerplate is doing before I dive in.

I have been going through the source and trying my best to understand whats happening, but searching through docs over and over can get boring/confusing quite quickly if you're someone who is just starting to learn / grasp all of the more advanced aspects of JS tooling.

**Comment by martin0852** [comment]
Did this grow any legs? This boilerplate is awesome and even though I’ve gotten quite comfortable with the react/redux part of it, I’m still lost when anything webpacky or electron main process pops up- would love to see an example of building a windows installer package and setting up auto-updates!!

**Comment by amilajack** [comment]
I'm still interested in doing this. I'm just trying to find a period of time where I have the time to do this

**Comment by martin0852** [comment]
Cool - I doubt there is, but if there’s any help I can give let me know!

**Comment by amilajack** [comment]
I'm considering taking donations for this project. It does take a signification amount of time to resolve peoples' issues/questions, fix bugs, add features, etc.

**Comment by talha131** [comment]
@amilajack You should. There is already a [backer](https://github.com/chentsulin/electron-react-boilerplate#backers) link in readme. Is there any issue with it? Are active maintainers not getting donations?

**Comment by goldylucks** [comment]
How about donations directed at specific documentation / features added?

I'm in touch with some electron based projects that might donate/pay to get features or targeted help/docs.

Regardless, this project is ridiculously awesome, keep up the good work!

**Comment by dbvrac** [comment]
I’d be willing to donate toward some new tutorial examples of applying this project. IE: Add x css gizmos with y middleware to communicate with z information source, like a weather app. I’ve spent too much time trying to discect the current eample set only to find out they’re outdated or too different from the current ERB.

**Comment by EricPeter** [comment]
Hi guys anyone help, am new to electron I need to embed a machine learning model in an electron desktop application

**Comment by jefffriesen** [comment]
I don't know if that's an ideal topic for a video series. But I would consider tensorflow.js: https://www.tensorflow.org/js. It works for both training and inference in the browser. If you have a trained model from a library like Keras, you can convert it to work in the browser.

**Comment by leeqingyoung** [comment]
Is the tutorial released?

**Comment by amilajack** [comment]
Not yet. Coming soon!

**Comment by geek4teck** [comment]
Any update on the release date yet?
